runtime/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/KafkaGrpcConfiguration.java

```diff
@@ -31,4 +41,56 @@ public KafkaGrpcConfiguration(
     {
         super(KAFKA_GRPC_CONFIG, config);
     }
+
+    public Supplier<String> groupIdSupplier()
+    {
+        return KAFKA_GROUP_ID.get(this)::get;
+    }
+
+    @FunctionalInterface
+    private interface GroupIdSupplier extends Supplier<String>
+    {
+    }
+
+    private static GroupIdSupplier decodeGroupId(
+        Configuration config,
+        String fullyQualifiedMethodName)
+    {
+        GroupIdSupplier supplier = null;
+
+        try
+        {
+            MethodType signature = MethodType.methodType(String.class);
+            String[] parts = fullyQualifiedMethodName.split("::");
+            Class<?> ownerClass = Class.forName(parts[0]);
+            String methodName = parts[1];
+            MethodHandle method = MethodHandles.publicLookup().findStatic(ownerClass, methodName, signature);
+            supplier = () ->
+            {
+                String value = null;
+                try
+                {
+                    value = (String) method.invoke();
+                }
+                catch (Throwable ex)
+                {
+                    LangUtil.rethrowUnchecked(ex);
+                }
+
+                return value;
+            };
+        }
+        catch (Throwable ex)
+        {
+            LangUtil.rethrowUnchecked(ex);
+        }
+
+        return supplier;
+    }
+
+    private static GroupIdSupplier defaultGroupId(
+        Configuration config)
+    {
+        return () -> String.format("zilla-%s-%s", "", "");
```

Comment:
Suggest `zilla:%s-%s` instead, probably more readable.
I'm thinking we'll want to do the same for `mqtt-kafka` and perhaps make it possible to prefix with `mqtt:` instead of `zilla:` if desired to make it easier to differentiate the consumer groups using standard kafka tools.